support-workers/src/typescript/services/salesforce.ts

```diff
@@ -140,68 +165,144 @@ export class SalesforceService {
 		giftRecipient: GiftRecipient | null,
 		user: User,
 	): Promise<SuccessfulUpsertResponse> | null {
```

Comment:
Not strictly related to this PR, but I wonder if it'd be nice to have the return value of this function be consistent and always return a promise (even if there's nothing to do?). E.g. return `Promise.resolve(null)` if there's nothing to do. That way callers can either `await` or chain handlers on without needing to check the return value first.